pkg/controllermanager/controller/project/project_control_reconcile.go

```diff
@@ -120,31 +126,56 @@ func (c *defaultControl) reconcile(project *gardencorev1beta1.Project, projectLo
 	var (
 		admins  []rbacv1.Subject
 		viewers []rbacv1.Subject
+		customs []map[string]interface{}
+
+		customRoleToSubjects = map[string][]rbacv1.Subject{}
+		customRoles          = sets.NewString()
 	)
 
 	for _, member := range project.Spec.Members {
-		if member.Role == gardencorev1beta1.ProjectMemberAdmin {
-			admins = append(admins, member.Subject)
-		}
-		if member.Role == gardencorev1beta1.ProjectMemberViewer {
-			viewers = append(viewers, member.Subject)
+		for _, role := range member.Roles {
+			if role == gardencorev1beta1.ProjectMemberAdmin || role == gardencorev1beta1.ProjectMemberOwner {
+				admins = append(admins, member.Subject)
+			}
+			if role == gardencorev1beta1.ProjectMemberViewer {
+				viewers = append(viewers, member.Subject)
+			}
+			if strings.HasPrefix(role, gardencorev1beta1.ProjectMemberCustomPrefix) {
+				customRoleToSubjects[role] = append(customRoleToSubjects[role], member.Subject)
+				customRoles.Insert(strings.TrimPrefix(role, gardencorev1beta1.ProjectMemberCustomPrefix))
+			}
 		}
 	}
 
+	for _, name := range customRoles.List() {
+		customs = append(customs, map[string]interface{}{
+			"name":     name,
+			"subjects": customRoleToSubjects[name],
```

Comment:
This won't work, `subjects` will always be empty, because you are using the custom prefix in the keys in `customRoleToSubjects`, but in `customRoles` the names won't have the prefix. So `customRoleToSubjects[name]` will always be `nil`.